### PR TITLE
Stack images overhaul

### PR DESCRIPTION
**Pull request by eivantsov**
What does this PR do:

* cleaner structure of images, use of eclipse/stack-base:tag images where possible. Makes it easier to introduce changes and test them
* use of openjdk in base images
* Ubuntu 16.04 as a base image for all ubuntu based stacks
* cleanup of redundant commands (entrypoints, cmds etc)
* passwordless user (ssh with a key tested)
* build script that builds and tags images
* clear naming and tagging pattern
* fix build errors (some images were last built a long time ago, and some links and even the entire repo)

@kaloyan-raev will provide a fresh PHP stack based on PHP7. SO, this will be merged after he pushes his changes.

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
I updated the PHP stack to PHP 7.0.

File: recipes/alpine_jdk8/Dockerfile [hunk 1]
```diff
@@ -86,13 +87,14 @@ RUN apk upgrade --update && \
     rm "/tmp/"* && \
     adduser -S user -h /home/user -s /bin/bash -G root -u 1000 && \
     echo "%root ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
-    PASS=$(openssl rand -base64 32) && \
-    echo -e "${PASS}\n${PASS}" | passwd user && \
+    usermod -p "*" user && \
+#    PASS=$(openssl rand -base64 32) && \
+#    echo -e "${PASS}\n${PASS}" | passwd user && \
```
Comment (skabashnyuk) [inline recipes/alpine_jdk8/Dockerfile]: remove if not used

File: recipes/alpine_jdk8/Dockerfile [hunk 2]
```diff
@@ -86,13 +87,14 @@ RUN apk upgrade --update && \
...
     chown -R user /home/user/
 
-RUN mkdir /usr/che \
-    && wget -qO /usr/che/che https://raw.githubusercontent.com/eclipse/che/master/che.sh \
-    && chmod +x /usr/che/che
+#RUN mkdir /usr/che \
+#    && wget -qO /usr/che/che https://raw.githubusercontent.com/eclipse/che/master/che.sh \
+#    && chmod +x /usr/che/che
```
Comment (skabashnyuk) [inline recipes/alpine_jdk8/Dockerfile]: remove if not used

**Comment by monaka** [comment]
eclipse/dotnet_core/Dockerfile has `FROM eclipse/debian_jre` that is removed by this PR.
Which is better, fixing dotnet_core or reverting eclipse/debian_jre ?

**Comment by TylerJewell** [comment]
We moved everything to openjdk to avoid issues with legal ambiguity on distributing JRE's within Docker images. Our lawyers made us do it!

**Comment by monaka** [comment]
I see. I'll send PR for dotnet_core/Dockerfile.

**Comment by eivantsov** [comment]
@monaka no need to do that. dotnet stack isn't going anywhere - this PR brings it back again https://github.com/eclipse/che-dockerfiles/pull/97

**Pull request merged**